patroni/ctl.py

```diff
@@ -1117,6 +1116,9 @@ def restart(cluster_name: str, group: Optional[int], member_names: List[str],
     content: Dict[str, Any] = {}
     if pending:
         content['restart_pending'] = True
+        members = [m for m in members if m.data.get('pending_restart', False)]
```

Comment:
I think this block should be before [if p_any:](https://github.com/patroni/patroni/pull/3226/files#diff-9dada8f509128404d7027246ac9622d2484fc77758c6a4af9dc7cc8c081d9fffR1108)